src/test/rgw/rgw_multi/tests.py

```diff
@@ -716,3 +717,51 @@ def test_zonegroup_remove():
 
     # validate the resulting period
     zonegroup.period.update(z1, commit=True)
+
+def test_null_version():
+    master_zg = realm.master_zonegroup()
+    zone_m = master_zg.master_zone
+    conn = get_zone_connection(zone_m, user.credentials)
+    bucket_name = gen_bucket_name()
+    bucket = conn.create_bucket(bucket_namee)
+
+    objnames = ['myobj1', 'myobj2', 'myobj3']
+    content = 'abcdefg'
+
+    bucket.configure_versioning(True)
+    zonegroup_meta_checkpoint(master_zg)
+
+    for i in range(0, 2):
+        k = new_key(zone_m, bucket, objnames[i])
+        k.set_contents_from_string(content)
+    for zone in master_zg.zones:
+        zone_bucket_checkpoint(zone, zone_m, bucket.name)
+    bucket.configure_versioning(False)
+    zonegroup_meta_checkpoint(master_zg)
+
+    for i in range(0, 3):
+        k = new_key(zone_m, bucket, objnames[i])
+        k.set_contents_from_string(content)
+    for zone in master_zg.zones:
+        zone_bucket_checkpoint(zone, zone_m, bucket.name)
+    rs = bucket.get_all_versions()
+    assert(len(rs) == 5)
+
+    bucket.delete_key(objnames[0], None, 'null')
+    bucket.delete_key(objnames[1])
+    bucket.delete_key(objnames[2])
+    rs = bucket.get_all_versions()
+    assert(len(rs) == 4)
+
+    for o in bucket.get_all_versions():
+        bucket.delete_key(o.name, None, o.version_id)
+    for zone in master_zg.zones:
+        zone_bucket_checkpoint(zone, zone_m, bucket.name)
+        check_bucket_eq(zone_m, zone, bucket)
+	cmd = ['ls', '-p', str(zone.name) + '.rgw.buckets.data']
+        olist_json, retcode = zone.cluster.rados(cmd)
+	if retcode != 0:
+            log.debug('list pool %s.rgw.buckets.data error', zone.name)
```

Comment:
note that `bash()` in `test_multi.py` has a `check_retcode` keyword that defaults to `True`, so we would hit an assertion there before the `zone.cluster.rados()` call returned a non-zero retcode

(p.s. i love that you're using `rados` to inspect the underlying objects - i hope to see more of this!)